djangocms_navigation/admin.py

```diff
@@ -464,17 +464,52 @@ def changelist_view(self, request, menu_content_id=None, extra_context=None):
         return super().changelist_view(request, extra_context)
 
     def response_change(self, request, obj):
-        url = reverse(
-            "admin:{}_menuitem_list".format(self.model._meta.app_label),
-            kwargs={"menu_content_id": request.menu_content_id},
-        )
+        msgstr = 'Menuitem "{}" was changed successfully.'.format(obj)
+        extras = ""
+        if '_addanother' in request.POST:
```

Comment:
A few things with this, it's missing translations, has repeated logic and has mixed quotes all over.

```python
    def response_change(self, request, obj):
        msg = _('Menuitem %(menuitem)s was changed successfully.') % {'menuitem': obj}
        endpoint_name = "menuitem_list"
        endpoint_kwargs={"menu_content_id": request.menu_content_id}

        if "_addanother" in request.POST:
 	    endpoint_name = "menuitem_add"
        elif "_continue'" in request.POST and self.has_change_permission(request):
            msg = _('Menuitem %(menuitem)s was changed successfully. You can edit it below') % {'menuitem': obj}
            endpoint_kwargs.update({"object_id": obj.id})

        url = reverse(
            "admin:{}_{}".format(self.model._meta.app_label, endpoint_name),
            kwargs=endpoint_kwargs,
        )

        self.message_user(request, msg, messages.SUCCESS)
```